js/models/BaseModel.js

```diff
@@ -121,17 +123,30 @@ export default class extends Model {
           const nestedData = attrs[nestedKey];
           const nestedInstance = this.attributes[nestedKey];
 
-          if (nestedInstance) {
-            if (nestedData) nestedInstance.set(nestedData);
-            delete attrs[nestedKey];
-          } else {
-            attrs[nestedKey] = new NestedClass(nestedData);
+          if (nestedData) {
+            if (nestedData instanceof NestedClass) {
+              attrs[nestedKey] = nestedData;
+            } else if (nestedInstance) {
+              nestedInstance.set(nestedData);
+              delete attrs[nestedKey];
+            } else {
+              attrs[nestedKey] = new NestedClass(nestedData);
+            }
           }
         });
       }
     }
 
-    return super.set(attrs, opts);
+    const superSet = super.set(attrs, opts);
+
+    // Since the standard change event doesn't properly take into
```

Comment:
An example best illustrates why this event is necessary... Let's say we have a Model with a nested Model, e.g.

```
const luckyLarry = new Human({
  eyeColor: 'blue',
  car: {
    make: 'Toyota',
    model: 'Corolla',
    year: 2006,
  }
});

luckyLarry.on('change', () => { // do something });
luckyLarry.get('car').on('change', () => { // do something });

luckyLarry.set('car', {
  year: 2008,
});
```

By default, the second change event will fire, but not the first, since the instance isn't changing. So, I've temporarily introduced a `someChange` event that will fire if any part of the model (nesting and all) changes, whereas you could still directly bind to nested instances if you need to know whether a specific nested attribute changed.

I've actually brainstormed and I think I have a good strategy to integrate this behavior into the standard change event (without modifying Backbone source!), but there's a little bit of work there to also make changedAttributes() and hasChanged() work as well, so this `someChange` event is a bridge to that happening at some point.